nuclear_simulator/simulator/state/state_manager.py

```
@@ -0,0 +1,1910 @@
+            provider: Component that provides state variables
+            category: Category name for organizing the provider's variables
+        """
+        # Register the provider
+        self.providers.append((provider, category))
+        
+        # Register all state variables from this provider
+        try:
+            state_variables = provider.get_state_variables()
+            self.registry.register_variables(state_variables)
+        except Exception as e:
+            warnings.warn(f"Failed to register state variables from provider {category}: {e}")
```

Ruff (0.11.9)

No explicit `stacklevel` keyword argument found

Set `stacklevel=2`

(B028)